ansible/roles_ocp_workloads/ocp4_workload_rhods_olm/tasks/workload.yml

```diff
@@ -0,0 +1,170 @@
+  vars:
+    _query: >-
+      [?contains(spec.clusterServiceVersionNames[0], 'rhods-operator')]
+  retries: 30
+  delay: 5
+  until:
+    - r_install_plans.resources | length > 0
+    - r_install_plans.resources | to_json | from_json | json_query(_query)
+
+- name: Set InstallPlan Name
+  set_fact:
+    ocp4_workload_rhods_olm_install_plan_name: "{{ r_install_plans.resources | to_json | from_json | json_query(query) }}"
+  vars:
+    query: >-
+      [?contains(spec.clusterServiceVersionNames[0], 'rhods-operator')].metadata.name|[0]
+
+- name: Get InstallPlan
+  k8s_info:
+    api_version: operators.coreos.com/v1alpha1
+    kind: InstallPlan
+    name: "{{ ocp4_workload_rhods_olm_install_plan_name }}"
+    namespace: "{{ ocp4_workload_rhods_olm_odh_namespace }}"
+  register: r_install_plan
+
+- name: Approve InstallPlan if necessary
+  when: r_install_plan.resources[0].status.phase is match("RequiresApproval")
+  k8s:
+    state: present
+    definition: "{{ lookup( 'template', './templates/installplan.j2' ) }}"
+
+- name: Get Installed CSV
+  k8s_info:
+    api_version: operators.coreos.com/v1alpha1
+    kind: Subscription
+    name: addon-managed-odh
+    namespace: "{{ ocp4_workload_rhods_olm_odh_namespace }}"
+  register: r_subscription
+  retries: 30
+  delay: 5
+  until:
+    - r_subscription.resources[0].status.currentCSV is defined
+    - r_subscription.resources[0].status.currentCSV | length > 0
+
+- name: Wait until CSV is Installed
+  k8s_info:
+    api_version: operators.coreos.com/v1alpha1
+    kind: ClusterServiceVersion
+    name: "{{ r_subscription.resources[0].status.currentCSV }}"
+    namespace: "{{ ocp4_workload_rhods_olm_odh_namespace }}"
+  register: r_csv
+  retries: 15
+  delay: 5
+  until:
+    - r_csv.resources[0].status.phase is defined
+    - r_csv.resources[0].status.phase | length > 0
+    - r_csv.resources[0].status.phase == "Succeeded"
+
+- name: Wait for jupyter deploymentconfigs to exist
+  k8s_info:
+    api_version: apps.openshift.io/v1
+    kind: DeploymentConfig
+    name: "{{ item }}"
+    namespace: "{{ ocp4_workload_rhods_olm_applications_namespace }}"
+  loop:
+    - "jupyterhub"
+    - "jupyterhub-db"
+  register: r_jh_dc
+  retries: 24
+  delay: 20
+  until:
+    - r_jh_dc.resources | length > 0
+
+- name: Wait for ready jupyter deploymentconfigs
+  k8s_info:
+    api_version: apps.openshift.io/v1
+    kind: DeploymentConfig
+    name: "{{ item }}"
+    namespace: "{{ ocp4_workload_rhods_olm_applications_namespace }}"
+  loop:
+    - "jupyterhub"
+    - "jupyterhub-db"
+  register: r_jh_dc
+  retries: 24
+  delay: 20
+  until:
+    - r_jh_dc.resources[0].status.readyReplicas is defined
+    - r_jh_dc.resources[0].status.readyReplicas | int >= 1
+
+- name: Update the RHODS config to allow all users
+  command: |
+    oc patch configmap/rhods-groups-config -n  redhat-ods-applications -p '{"metadata": {"labels": {"opendatahub.io/modified":"true"}}, "data":{"allowed_groups":"system:authenticated"}}'
```

Comment:
Please use `k8s` module rather than `command`.
```
- name: Update the RHODS config to allow all users
  k8s:
    api_version: v1
    kind: ConfigMap
    namespace: redhat-ods-applications
    name: rhods-groups-config
    definition:
      labels:
        opendatahub.io/modified: "true"
      data:
        allowed_groups: system:authenticated
```

Reply:
@jkupferer is the default behavior for the k8s module to patch existing objects?

Reply:
Yes